dGame/dComponents/DestroyableComponent.cpp

```diff
@@ -765,20 +766,16 @@ void DestroyableComponent::Smash(const LWOOBJID source, const eKillType killType
 			uint64_t coinsTotal = character->GetCoins();
 
 			if (coinsTotal > 0) {
-				uint64_t coinsToLoose = 1;
+				const uint64_t minCoinsToLose = dZoneManager::Instance()->GetWorldConfig()->coinsLostOnDeathMin;
+				const uint64_t maxCoinsToLose = dZoneManager::Instance()->GetWorldConfig()->coinsLostOnDeathMax;
+				const float coinPercentageToLose = dZoneManager::Instance()->GetWorldConfig()->coinsLostOnDeathPercent;
 
-				if (coinsTotal >= 200) {
-					float hundreth = (coinsTotal / 100.0f);
-					coinsToLoose = static_cast<int>(hundreth);
-				}
-
-				if (coinsToLoose > 10000) {
-					coinsToLoose = 10000;
-				}
+				uint64_t coinsToLose = std::max(static_cast<uint64_t>(coinsTotal * coinPercentageToLose), minCoinsToLose);
+				coinsToLose = std::min(maxCoinsToLose, coinsToLose);
 
-				coinsTotal -= coinsToLoose;
+				coinsTotal -= coinsToLose;
 
-				LootGenerator::Instance().DropLoot(m_Parent, m_Parent, -1, coinsToLoose, coinsToLoose);
+				LootGenerator::Instance().DropLoot(m_Parent, m_Parent, -1, coinsToLose, coinsToLose);
 				character->SetCoins(coinsTotal, eLootSourceType::LOOT_SOURCE_PICKUP);
```

Comment:
This seems like a bit of a change in logic. Should the player lose a linear amount of coins immediately? Not sure where that initial 200 number came from, but this will act differently from how it currently does. Also as a redundancy check, should the above if-statement be conditioned on `coinsTotal  >= coinsLostOnDeathMin` as to prevent accidental integer overflows if the value is changed.

Reply:
That's the problem with a magic number like 200.  We have no idea where that value came from. There should be a floor that you have at least 1 coin to lose yes and ill implement that, but this should ensure that you lose at least 1 coin.  Unless there's a missing worldconfig option, this logic is correct with the floor of coins to lose

Reply:
I've pushed changes to change the if conditional.  Unsure what you mean about the first question, however the existence of the minimum coins to lose to me implies that if you have money > min coins to lose, then you lose at least that amount of currency. Tested that dying with 2 1 and 0 coins nets a loss of 1 1 and 0 coins respectively.

Reply:
the 200 check is completely arbitrary.  The minimum dropped coins is based on `coins_lost_on_death_min` as changing this value to 15 in the cdclient.fdb resulted in 15 coins being dropped and having less than 15 coins resulted in the client dropping all the coins the player had.  This can be tested by turning the flag `bSpawnLoot` to true in the SendDie GameMessage and not handling any server coin drop logic at all.